packages/entities/entities-plugins/src/components/PluginEntityForm.vue

```diff
@@ -0,0 +1,734 @@
+      // scope and top level fields
+      if (props.record.consumer_id || props.record.service_id || props.record.route_id || props.record.consumer_group_id) {
+        updateModel({
+          service_id: props.record.service_id,
+          route_id: props.record.route_id,
+          consumer_id: props.record.consumer_id,
+          consumer_group_id: props.record.consumer_group_id,
+          enabled: props.record.enabled,
+        })
+      }
+
+      // main plugin configuration
+      updateModel(props.record.config, 'config')
+    }
+  }
+
+  // scoping logic, convert _ to - for form model
+  // Check if incoming field exists in current model and if so update
+  if (entityIdField.value && props.entityId && props.schema) {
+    const updateFields: Record<string, any> = {}
+    const key = entityIdField.value === 'consumer_group_id' ? 'consumer_group-id' : JSON.parse(JSON.stringify(entityIdField.value).replace('_', '-'))
+
+    // ex. set consumer-id: <entityId>
+    if (Object.prototype.hasOwnProperty.call(formModel, key)) {
+      updateFields[key] = props.entityId
+    }
+
+    updateModel(updateFields)
+  }
+
+  // Check if entity field exists in current model and if so update
+  // credentials don't recognize field with -id, so for now set it like
+  // ex. consumer: <entityId>
+  // we'll fix this on submit
+  if (props.entityId && props.schema && props.credential) {
+    const updateFields: Record<string, any> = {}
+    if (Object.prototype.hasOwnProperty.call(formModel, props.entityType)) {
+      updateFields[props.entityType] = props.entityId
+    }
+
+    updateModel(updateFields)
+  }
+
+  loading.value = false
+}
+
+watch(loading, (newLoading) => {
+  emit('loading', newLoading)
+})
+
+// if the schema changed we've got to start over and completely rebuild the form model
+watch(() => props.schema, (newSchema) => {
+  const form: Record<string, any> = parseSchema(newSchema)
+
+  Object.assign(formModel, form.model)
+
+  formSchema.value = { fields: formSchema.value?.fields?.map((r: Record<string, any>) => { return { ...r, disabled: r.disabled || false } }) }
+  Object.assign(originalModel, JSON.parse(JSON.stringify(form.model)))
+  sharedFormName.value = getSharedFormName(form.model.name)
+
+  initFormModel()
+}, { immediate: true, deep: true })
+
+onBeforeMount(() => {
+  form.value = parseSchema(props.schema)
+
+  Object.assign(formModel, form.value?.model || {})
+  formSchema.value = form.value?.schema || {}
+
+  initFormModel()
+})
+</script>
+
+<style lang="scss" scoped>
+.kong-ui-entities-plugin-form {
+  width: 100%;
+
+  .fade-enter-active,
+  .fade-leave-active {
+    transition: opacity 0.5s;
+  }
+
+  .fade-enter-from, .fade-leave-to {
+    opacity: 0;
+  }
+
+  :deep(.vue-form-generator) {
+    & > fieldset {
+      border: none;
+      padding: $kui-space-0;
+    }
+
+    .field-switch .field-wrap label {
+      .label {
+        background-color: $kui-color-background-neutral-weak;
```

Comment:
This line makes the switch gray even when it is checked. Is it intentional?
<img width="443" alt="image" src="https://github.com/Kong/public-ui-components/assets/10095631/39d95235-5185-4603-a600-1f900bd9540c">